include/seqan3/alphabet/union_alphabet.hpp

```diff
@@ -99,6 +78,88 @@ constexpr auto at_least_type() {
     }
 }
 
+template <size_t size>
+using at_least_type_t = decltype(detail::at_least_type<size>());
+
+template <typename ...alphabet_types>
+constexpr auto alphabet_prefix_sum_sizes()
+{
+    constexpr size_t value_size = ((size_t)alphabet_types::value_size + ... + 0);
+    using integral_t = at_least_type_t<value_size>;
+
+    constexpr size_t N = sizeof...(alphabet_types) + 1;
+    using array_t = std::array<integral_t, N>;
+
+    array_t prefix_sum{0, alphabet_types::value_size...};
+    for(auto i = 1u; i < N; ++i)
+        prefix_sum[i] += prefix_sum[i-1];
+
+    return prefix_sum;
+}
+
+namespace union_alphabet
```

Comment:
Since we don't indent the namespaces, it's difficult to keep track.
I would propose to never declare them in nested fashion, and instead do
```c++
namespace seqan3::detail
{

} // namespace seqan3::detail

namespace seqan3::detail::union_alphabet
{

} // namespace seqan3::detail::union_alphabet

namespace seqan3
{

} // namespace seqan3
```

I will also change this in the other files accordingly. 

PS: Yes, I know, we need to set up documentation on how to do this correctly.